golem-worker-bridge/src/register.rs

```diff
@@ -194,30 +232,145 @@ impl RegisterApiDefinition for RedisApiRegistry {
+                move |transaction| async move {
+                    transaction.del(definition_key).await?;
+                    transaction
+                        .srem(version_set_key.clone(), vec![api_version])
+                        .await?;
+                    transaction.scard(version_set_key).await?;
+
+                    Ok(transaction)
+                }
+            })
             .await
             .map_err(|e| ApiRegistrationError::InternalError(e.to_string()))?;
+
+        // If the definition was deleted and there are no more versions, remove the definition from the all_api_definitions set.
+        if definition_delete > 1 && num_versions == 0 {
+            let all_apis_key = get_all_apis_set_redis_key();
+            let api_version = api_id.version.0.clone();
+            self.pool
+                .with("persistence", "delete_definition")
+                .srem(all_apis_key, vec![api_version])
+                .await
+                .map_err(|e| {
+                    tracing::error!(
+                        "Failed to remove definition from all_api_definitions set: {}",
+                        e
+                    );
+                    ApiRegistrationError::InternalError(e.to_string())
+                })?;
+        }
+
         Ok(definition_delete > 0)
     }
 
     async fn get_all(&self) -> Result<Vec<ApiDefinition>, ApiRegistrationError> {
-        unimplemented!("get_all")
+        let all_apis: Vec<ApiDefinitionId> = {
+            let result: Vec<String> = self
+                .pool
+                .with("persistence", "get_all_definitions")
+                .smembers(get_all_apis_set_redis_key())
+                .await
+                .map_err(|e| ApiRegistrationError::InternalError(e.to_string()))?;
+
+            result.into_iter().map(ApiDefinitionId).collect()
+        };
+
+        let all_versions: Vec<(ApiDefinitionId, Vec<String>)> = {
+            let futures = all_apis.into_iter().map(|api_id| async move {
+                let version_set_key = get_api_definition_redis_key_version_set(&api_id);
+                let versions: Vec<String> = self
+                    .pool
+                    .with("persistence", "get_all_definitions")
+                    .smembers(version_set_key)
+                    .await
+                    .map_err(|e| ApiRegistrationError::InternalError(e.to_string()))?;
+                Ok((api_id, versions))
+            });
+
+            futures::future::try_join_all(futures).await?
+        };
+
+        let all_definitions: Vec<Option<ApiDefinition>> = {
+            let futures = all_versions
+                .into_iter()
+                .flat_map(|(api_id, versions)| {
+                    versions.into_iter().map(move |version| ApiDefinitionKey {
+                        id: api_id.clone(),
+                        version: Version(version),
+                    })
+                })
+                .map(|api_id| async move { self.get(&api_id).await })
```

Comment:
calling `GET` for each api definition in iteration, is not very good idea 
i think that redis https://redis.io/commands/mget/  should be able to fetch multiple keys with one call

Reply:
yeah I mentioned that in PR description. I will implement

Reply:
nice, sorry, i missed description , but i think it will be good to implement it here or in following PR ...